wpilibj/src/main/java/edu/wpi/first/wpilibj/trajectory/Trajectory.java

```diff
@@ -26,6 +26,14 @@
   private final double m_totalTimeSeconds;
   private final List<State> m_states;
 
+  /**
+   * Constructs an empty trajectory.
+   */
+  public Trajectory() {
+    m_states = new ArrayList<>();
+    m_totalTimeSeconds = 0.0;
```

Comment:
In Java, doesn't this default to 0?

Reply:
I believe it doesn't default to zero for final member variables.

Reply:
Yup – final fields of a class that have no default value and must be explicitly initialized just once before a class constructor finished.

https://docs.oracle.com/javase/specs/jls/se7/html/jls-8.html#jls-8.3.1.2